core/lb.go

```diff
@@ -147,8 +146,9 @@ func (lb *LoadBalancingTranscoder) leastLoaded() string {
 }
 
 type transcoderParams struct {
-	md  *SegTranscodingMetadata
-	res chan struct {
+	logCtx context.Context
```

Comment:
This looks fishy. In general Go doc [does not recommend storing context in any structs](https://pkg.go.dev/context).
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

I understand that `transcoderParams` are actually for us the request params, but still... 

It makes me think that we should maybe have an abstraction like `clog` struct which we store inside the context. Then, pass only this struct into `transcoderParams` (and to all other places that do not respect the `context` (i.e. does not stop working when the context is canceled) ).

Reply:
Yes, it is looks not ideal, but I think it's better than the other option. Introducing separate structure for logging data will lead to a lot of wrapping/unwrapping code, and for what? To satisfy one sentence in context package's documentation? 
Context is useful because it is a black box, and functions can pass it around without knowing what's inside.

Reply:
> In general Go doc does not recommend storing context in any structs.

@leszko I think that's a rule that needs to be interpreted with a grain of salt as well. At least I interpret it as not storing ctxs in a long-living struct that will be shared among multiple short-lived operations, each of these operations (naturally) with their own respective ctxs. 

I think in the case of a struct only used to pass parameters around it is very much an exception of that, as it is mostly a syntax sugar for not repeating the parameters multiple times in several places, but all of those within one of those short-live operations with the same ctx. (I am assuming that this is the use case for this params struct, since it is also so short with only 3 fields). An example of that is go's own `http.Request` object which can hold a ctx for that specific request to be made: https://pkg.go.dev/net/http#NewRequestWithContext 

Another exception for that rule I think are "background jobs" which should be cancellable with a context passed on creation of the job.  In that case, you could in theory pass the context around in every internal struct that needs access to it, but I think it's also valid to store the job context in the struct that represents the job itself, and access it anywhere you need it. One example of those is the RabbitMQ Streams consumer [here](https://github.com/livepeer/livepeer-data/blob/995038a18eed07b6b61c02983f571250c393f5ea/pkg/event/stream_consumer.go#L23) (it has an internal `done` channel instead, but could just as well be an external `context.Context` for the same result).

Reply:
Yeah makes sense! I agree that having a separate context that is used for nothing but logging would be an anti-patttern. IMO, there should be only one `ctx` passed around [1], which in some places is used only for cancellation in case no logging is done, in others is used only for logging in case cancellation is not supported, and in the ideal cases are used for both. With that in mind, this should be renamed to just `ctx` and if it's built out of thin air just for logging it should be from a `context.TODO()` to be fixed later with a real context, not a `context.Background()` (not sure if that's happening anywhere right now).

[1] The only exception case I've found for this rule is when you do actually have multiple hierarchical cancellation contexts for the current code. For example if you have a server, which has it's own `ctx` for shutting down, while each request also has it's own `ctx`. You might use both, or better yet "merge" them into a single `ctx` that you pass around further down, using the same example as before, like this: https://github.com/livepeer/livepeer-data/blob/ed2b74fc9a2124032c05b11addfa4ebb6c013aab/pkg/event/stream_consumer.go#L199

Reply:
Yeah, agree 🤝

Reply:
@leszko @victorges 
I've renamed logCtx to `ctx` in most places, now it's ready for review.